fv3_cap.F90

```diff
@@ -340,92 +340,13 @@ subroutine InitializeAdvertise(gcomp, rc)
     call ESMF_ConfigGetAttribute(config=CF, value=dt_atmos, label ='dt_atmos:',   rc=rc)
     call ESMF_ConfigGetAttribute(config=CF, value=nfhmax,   label ='nhours_fcst:',rc=rc)
     if(mype == 0) print *,'af nems config,dt_atmos=',dt_atmos,'nfhmax=',nfhmax
-    call ESMF_TimeIntervalSet(timeStep,s=dt_atmos,rc=rc)
-    call ESMF_ClockSet(clock_fv3,timeStep=timeStep, rc=rc)
-!
-!------------------------------------------------------------------------
-! may need to set currTime for restart
-!
-    call ESMF_ClockGet(clock_fv3, currTime=currTime,  StartTime=startTime,    &
-                       RunDuration=RunDuration, rc=rc)
-    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
-
-    stopTime = startTime + RunDuration
-
-! *** read restart time from restart file
-    do i=751,899
-       inquire(i, opened=opened)
-       if(.not. opened)then
-         io_unit = i
-         exit
-       endif
-    enddo
-!
-    date = 0 ; date_init = 0
-    force_date_from_configure = .true.
-!
-    open(unit=io_unit, file=trim('INPUT/coupler.res'),status="old",err=998 )
-    read (io_unit,*,err=999) calendar_type
-    read (io_unit,*) date_init
-    read (io_unit,*) date
-    close(io_unit)
-    force_date_from_configure = .false.
-!
-    if(date(1) == 0 .and. date_init(1) /= 0) date = date_init
-    if(mype == 0) print *,'bf clock_fv3,date=',date,'date_init=',date_init
 
-    call ESMF_VMbroadcast(vm, date, 6, 0)
-    call ESMF_TimeSet(time=currTime,yy=date(1),mm=date(2),dd=date(3),h=date(4), &
-                      m=date(5),s=date(6),rc=rc)
+    call ESMF_TimeIntervalSet(timeStep, s=dt_atmos, rc=rc)
     if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
-999 continue
-998 continue
-!    if(mype==0) print *,'final date =',date,'date_init=',date_init
-
-!reset currTime in clock
-    call ESMF_ClockSet(clock_fv3, currTime=currTime, startTime=startTime,  &
-                       stopTime=stopTime, timeStep=timeStep, rc=rc)
 
+    call ESMF_ClockSet(clock_fv3, timeStep=timeStep, rc=rc)
```

Comment:
clock_fv3 is created from clock, which is the driver clock, the clock variable should have time step defined in run_sequence, which is not the fv3 internal time step. I am not clear why we need to redefine the timestep for driver clock here,. If we need a clock with timeStep of dt_atmos, maybe we can get the model clock directly instead of using the driver clock?

Reply:
this line does not redefine the time step of driver clock. clock_fv3 is a separate copy, local variable. I can try to use the model clock. Let me make a change and test it.

Reply:
Right, it does not change driver clock. What I am saying is that we probably just need the model clock.

Reply:
When I try to use modelClock and use it as clock_fv3:
```diff
     if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
 
     ! query for importState and exportState
-    call NUOPC_ModelGet(gcomp, driverClock=clock, importState=importState, exportState=exportState, rc=rc)
+    call NUOPC_ModelGet(gcomp, modelClock=clock_fv3, importState=importState, exportState=exportState, rc=rc)
     if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
 
     call ESMF_AttributeGet(gcomp, name="ProfileMemory", value=value, defaultValue="true", &
```
I get this run time error:
```
20211223 153550.988 ERROR            PET000 ESMF_Comp.F90:1519 ESMF_CompGet Invalid object  - requested Clock object is not present. 
20211223 153550.988 ERROR            PET000 ESMF_GridComp.F90:1129 ESMF_GridCompGet Invalid object  - Internal subroutine call returned Error
20211223 153550.988 ERROR            PET000 ATM:src/addon/NUOPC/src/NUOPC_ModelBase.F90:2742 Invalid object  - Passing error in return code
20211223 153550.988 ERROR            PET000 ATM:src/addon/NUOPC/src/NUOPC_Model.F90:195 Invalid object  - Passing error in return code
20211223 153550.988 ERROR            PET000 fv3_cap.F90:218 Invalid object  - Passing error in return code                           
20211223 153550.988 ERROR            PET000 ATM:src/addon/NUOPC/src/NUOPC_ModelBase.F90:700 Invalid object  - Passing error in return code
20211223 153550.988 ERROR            PET000 EARTH Grid Comp:src/addon/NUOPC/src/NUOPC_Driver.F90:2577 Invalid object  - Phase 'IPDvXp01' Initialize for modelComp 2: ATM did not return ESMF_SUCCESS
20211223 153550.988 ERROR            PET000 EARTH Grid Comp:src/addon/NUOPC/src/NUOPC_Driver.F90:1286 Invalid object  - Passing error in return code
20211223 153550.988 ERROR            PET000 EARTH Grid Comp:src/addon/NUOPC/src/NUOPC_Driver.F90:457 Invalid object  - Passing error in return code
20211223 153550.988 ERROR            PET000 UFS.F90:398 Invalid object  - Aborting UFS                                               
20211223 153550.988 INFO             PET000 Finalizing ESMF  
```                                                                                                                                                        
Looks like modelClock is not (yet) defined at this moment.

Reply:
Thanks for the testing. I guess the model clock is not available at this time. Let's keep driver clock.

Reply:
Since `clock_fv3` is only used in init phases of fcst and write grid components, and as far as I can see we only use start time and curr time in the init phases, not time step, maybe we can just pass driver clock. Let me test this.

Reply:
Maybe you can use "ESMF_GridCompGet(gcomp, clock=clock, rc=rc)" to get model clock, instead of NUOPC_ModelGet?

Reply:
Is that clock going to be the same as modelClock, if we know that model clock (with correct time step) is not yet available?

Reply:
Not sure, it's not clear to me when the ModelSetRunClock is called. We should be able to get the modelClock after ModelSetRunClock

Reply:
Sorry, please ignore the ESMF_GridCompGet clock, I saw dt_atmos is read after NUOPC_ModelGet

Reply:
Just using driveClock seems to work. rt.sh is still running but so far all tests passed.

Reply:
Added in https://github.com/NOAA-EMC/fv3atm/pull/450/commits/5ab09c5a7e2de29c322cfe6af86f9d6ae367df66